rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,424 @@
+                network = 'arb-mainnet'
+            elif asset.chain_id == ChainID.BINANCE:
+                network = 'bnb-mainnet'
+            elif asset.chain_id == ChainID.AVALANCHE:
+                network = 'avax-mainnet'
+            elif asset.chain_id == ChainID.ZKSYNC_ERA:
+                network = 'zksync-mainnet'
+            elif asset.chain_id == ChainID.ETHEREUM:
+                network = 'eth-mainnet'
+            elif asset.chain_id == ChainID.POLYGON_ZKEVM:
+                network = 'polygonzkevm-mainnet'
+            elif asset.chain_id == ChainID.SCROLL:
+                network = 'scroll-mainnet'
+            elif asset.chain_id == ChainID.GNOSIS:
+                network = 'gnosis-mainnet'
+            elif asset.chain_id == ChainID.BASE:
+                network = 'base-mainnet'
+            elif asset.chain_id == ChainID.FANTOM:
+                network = 'fantom-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_NOVA:
+                network = 'arbnova-mainnet'
+            else:
+                raise UnsupportedAsset(asset.identifier)
+
+            return {'network': network, 'address': asset.evm_address}
+
+        # Alchemy team recommends using standard ticker symbols (e.g. ETH)
+        # over coingecko ids (e.g. ethereum) for better reliability.
+        return asset.symbol
+
+    @staticmethod
+    def _deserialize_current_price(
+            result: list[dict[str, Any]],
+            asset_identifier: str | dict[str, str],
+            from_asset: Asset,
+            to_asset: Asset,
+    ) -> Price:
+        """Deserializes current price data from Alchemy API response.
+
+        If price data is missing, invalid, or there's an error, ZERO_PRICE is returned.
+        Otherwise, the deserialized price is returned.
+        """
+        for entry in result:
+            if (
+                (isinstance(asset_identifier, str) and entry['symbol'] != asset_identifier) or
+                (
+                    isinstance(asset_identifier, dict) and
+                    entry['network'] != asset_identifier['network'] and
+                    entry['address'] != asset_identifier['address']
+                )
+            ):
+                continue
+
+            if 'error' in entry:
+                log.warning(
+                    f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}. '  # noqa: E501
+                    f'API returned error: {entry["error"]}',
+                )
+                return ZERO_PRICE
+
+            if not (prices := entry['prices']):
+                log.warning(
+                    f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}, '  # noqa: E501
+                    f'but received no prices in response',
+                )
+                return ZERO_PRICE
+
+            for price in prices:
+                if price['currency'] != 'usd':
+                    continue
```

Comment:
what other currency is in there?

Reply:
only usd. however, it's an array of prices so doing a `prices[0]['value']` increases the risk of surprise if they add a new fiat equivalent